### PR TITLE
Add push option in createOutputChannel API

### PR DESCRIPTION
**Pull request by sandy081**
After implementing Output channels to log into files directly from extension host and let the renderer listen to file changes, a delay got introduced between adding the data and it is shown in the UI. It might not be expected for some output channels who intend to show real time data. This PR introduces an option `push` that extension authors can use while creating an output channel to decide whether to push the data and show in the UI immediately.

https://github.com/Microsoft/vscode/issues/57618

### CODE REVIEW

**Comment by jrieken** [comment]
Alternative ideas for the parameter name are `force`, `live`, `noBuffer`, `inMemory`, or something that expresses this is for logging.

**Comment by sandy081** [comment]
I liked `live` option out of all as it also goes well with what we describes about the output channel. Also how about `noDelay`?

```
Creates a new [output channel](#OutputChannel) with the given name.
		 * By default there will be a delay in showing the appended data in the UI.
		 * Pass `true` for `live` argument to show the data immediately.
		 *
		 * *NOTE* Default output channels help in improving the performance of VS Code
```

**Comment by sandy081** [comment]
@jrieken Any updates on this. I would like to change the second parameter to options object so that it can be extensible in future

`options: {noDelay: boolean}`

**Comment by sandy081** [comment]
I think `noBuffer` also has similar sense of `noDelay`. `force` might be good one to use. Updated the PR.

File: src/vs/vscode.d.ts [hunk 1]
```diff
@@ -6221,11 +6221,16 @@ declare module 'vscode' {
...
+		 * By default the UI will not show the appended data immediately.
+		 * To force UI to show the data immediately pass options with `force` parameter set to `true`.
+		 *
+		 * *NOTE* Default output channels help in improving the performance of VS Code.
 		 *
 		 * @param name Human-readable string which will be used to represent the channel in the UI.
+		 * @param options Optional options to control how the channel will be created.
 		 */
-		export function createOutputChannel(name: string): OutputChannel;
+		export function createOutputChannel(name: string, options?: { force?: boolean }): OutputChannel;
```
Comment (jrieken) [inline src/vs/vscode.d.ts]: why an object? are there plans for more options?

File: src/vs/vscode.d.ts [hunk 2]
```diff
@@ -6221,11 +6221,16 @@ declare module 'vscode' {
 		export function createInputBox(): InputBox;
 
 		/**
-		 * Create a new [output channel](#OutputChannel) with the given name.
+		 * Creates a new [output channel](#OutputChannel) with the given name.
+		 * By default the UI will not show the appended data immediately.
```
Comment (jrieken) [inline src/vs/vscode.d.ts]: Hm, we should sell this better... How about: "Output channels use a separate transport mechanism when appending data. This helps to reduce CPU load on the UI process but also means that the output channel UI updates with a small delay. Use the `force`-flag to enforce immediate updates at the cost of higher CPU usage"

File: src/vs/vscode.d.ts (hunk 1 above)
Comment (sandy081) [reply src/vs/vscode.d.ts]: Yeah, there is a request to set a lang id for channels. If so it has to be set while creating

File: src/vs/vscode.d.ts (hunk 2 above)
Comment (sandy081) [reply src/vs/vscode.d.ts]: I like it. You have better marketing skills 👍 😄

**Comment by sandy081** [comment]
Updated

**Review by jrieken — APPROVED** [review]
🚢 it

**Pull request merged**